### PR TITLE
fixed 8535Q

### PR DESCRIPTION
**Pull request by willLPD2I**
Allowed children who died to not have a birth date

### CODE REVIEW

**Comment by tab1tha** [comment]
Oh apologies, I have read it again.
The first part of the rule doesn't allow this. A child who died should have been born.
We can't let it pass when the birthdate is missing.
There might be something else causing the bug.

File: cin_validator/rules/cin2022_23/rule_8535Q.py [hunk 1]
```diff
@@ -31,12 +31,10 @@ def validate(
...
     # Return rows where DOB is prior to DOD
     condition1 = df[PersonBirthDate] > df[PersonDeathDate]
-    # Return rows with no DOB
-    condition2 = df[PersonBirthDate].isna()
 
     # df with all rows meeting the conditions
-    df_issues = df[condition1 | condition2].reset_index()
-
+    df_issues = df[condition1].reset_index()
+    print(df_issues)
```
Comment (tab1tha) [inline cin_validator/rules/cin2022_23/rule_8535Q.py:37]: to delete

**Review by tab1tha — CHANGES_REQUESTED** [review]
The bug behaviour is worth a second look. 
The rule was already implemented as prescribed. I wonder why it doesn't work.

**Comment by willLPD2I** [comment]
> Oh apologies, I have read it again. The first part of the rule doesn't allow this. A child who died should have been born. We can't let it pass when the birthdate is missing. There might be something else causing the bug.

@tab1tha Not true, not all pregnancies are carried fully to term. In these instances, the Expected DOB is after the date of death.

File: cin_validator/rules/cin2022_23/rule_8535Q.py (hunk 1 above)
Comment (willLPD2I) [reply cin_validator/rules/cin2022_23/rule_8535Q.py:37]: Oh no, not the print 🤦

**Comment by tab1tha** [comment]
> > Oh apologies, I have read it again. The first part of the rule doesn't allow this. A child who died should have been born. We can't let it pass when the birthdate is missing. There might be something else causing the bug.
> 
> @tab1tha Not true, not all pregnancies are carried fully to term. In these instances, the Expected DOB is after the date of death. Unless, as it's a query, we still want to flag this.

There is another variable for unborn babies, ExpectedPersonBirthDate. This rule only interacts with PersonBirthDate which is for children who have been born.
However, **if** you notice that the children who are wrongly flagged all have an ExpectedPersonBirthDate and no PersonBirthDate (i.e unborn), a filter can be added before the issue locations are returned. The filter should exclude any children in the list who have an ExpectedPersonBirthDate.

**Comment by willLPD2I** [comment]
> > > Oh apologies, I have read it again. The first part of the rule doesn't allow this. A child who died should have been born. We can't let it pass when the birthdate is missing. There might be something else causing the bug.
> > 
> > 
> > @tab1tha Not true, not all pregnancies are carried fully to term. In these instances, the Expected DOB is after the date of death. Unless, as it's a query, we still want to flag this.
> 
> There is another variable for unborn babies, ExpectedPersonBirthDate. This rule only interacts with PersonBirthDate which is for children who have been born. However, **if** you notice that the children who are wrongly flagged all have an ExpectedPersonBirthDate and no PersonBirthDate (i.e unborn), a filter can be added before the issue locations are returned. The filter should exclude any children in the list who have an ExpectedPersonBirthDate.

They all have expected birth dates and no birth date meaning they died before they were born, so yes, I can add a filter for this and still keep the check for no birth date then

**Review by tab1tha — APPROVED** [review]

**Pull request merged**